utils/fixed_decimal/src/decimal.rs

```diff
@@ -495,6 +495,145 @@ impl FixedDecimal {
         self.check_invariants();
     }
 
+    fn n_magnitude(&self) -> i16 {
+        if self.digits.len() == 0 {
+            return self.magnitude;
+        }
+
+        self.magnitude - (self.digits.len() as i16 - 1)
```

Comment:
Issue: refer to the `pad_right` function for the correct logic for this.

```rust
let bottom_magnitude = (self.magnitude as i32 - self.digits.len() as i32 + 1) as i16;
```

Suggestion: name this function `bottom_magnitude()`, and then adopt it everywhere that this operation is performed in this file.

Reply:
actually, we have a function called: nonzero_magnitude_right which return the needed value

Reply:
yes, I added it in #1924

Reply:
Cool 😎